src/Compilers/VisualBasic/Portable/Parser/ParseStatement.vb

```diff
@@ -434,6 +434,9 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Syntax.InternalSyntax
             GetNextToken()
 
             Dim statement = SyntaxFactory.ElseStatement(elseKeyword)
+            If Not Context.IsLineIf AndAlso Not CurrentToken.IsEndOfLine Then
+                ReportSyntaxError(elseKeyword, ERRID.ERR_ExpectedEOS)
```

Comment:
This line is hit, but I can't see any diagnostics. I don't know what could be the issue.